tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs

```diff
@@ -1,11 +1,73 @@
 ﻿// ReSharper disable once CheckNamespace
 namespace Nox.Types.Tests.Types;
 
-public class InternetDomain
+public class InternetDomainTests
 {
+    [Theory]
+    [InlineData("example.com")]
+    [InlineData("example.org")]
+    [InlineData("example1.net")]
+    [InlineData("example.co.uk")]
+    [InlineData("example.com.au")]
+    public void InternetDomain_WithValidDomain_ShouldBeValid(string validDomain)
+    {
+       
+        // Arrange & Act
+        var internetDomain = InternetDomain.From(validDomain);
+
+        // Assert
+        Assert.NotNull(internetDomain);
+        Assert.Equal(validDomain, internetDomain.Value);
+    }
+
+    [Theory] 
+    [InlineData(null)]
+    [InlineData("")]
+    public void InternetDomain_WithNullOrEmptyDomain_ShouldBeInvalid(string invalidDomain)
+    {
+        // Arrange & Act
+        var exception = Assert.Throws<TypeValidationException> (() => _ = InternetDomain.From(invalidDomain));
+
+        // Assert
+        Assert.NotNull(exception);
+        Assert.Equal("Could not create a Nox InternetDomain type with an empty value.", exception.Errors.First().ErrorMessage);
+        
+    }
+    
+    
+    [Theory] 
+    [InlineData("example")]
+    [InlineData("example.123")]
+    [InlineData("example.org.1234")]
+    public void InternetDomain_WithInvalidDomain_ShouldBeInvalid(string invalidDomain)
+    {
+        // Arrange & Act
+        var exception = Assert.Throws<TypeValidationException>(() => _ = InternetDomain.From(invalidDomain));
+
+        // Assert
+        Assert.NotNull(exception);
+        Assert.Equal($"Could not create a Nox InternetDomain type with unsupported value '{invalidDomain}'.", exception.Errors.First().ErrorMessage);
+    }
+    
+    [Fact]
+    public void Nox_InternetDomain_Equality_Tests()
```

Comment:
IMO starting with Nox_ does not bring additional value, what do you think?

Compare_Same_InternetDomians_WithEqual_IsTrue for name, what do you thinh? Scenario, Methods, expected result? 

https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#best-practices

Reply:
Agree with you. I updated the method name.